service_container/3.3-di-changes.rst

```diff
@@ -468,6 +470,59 @@ Many autoconfigured tags have an optional priority. If you need to specify a pri
 (or any other optional tag attribute), no problem! Just :ref:`manually configure your service <services-manually-wire-args>`
 and add the tag. Your tag will take precedence over the one added by auto-configuration.
 
+5) Auto-configure with _instanceof
+----------------------------------
+
+And the final big change is ``_instanceof``. It acts as a default definition
+template (see `service-33-default_definition`_), but only for services whose
+class matches a defined one.
+
+This can be very useful when many services share some tag that cannot be
+inherited from an abstract definition:
+
+.. configuration-block::
+
+    .. code-block:: yaml
+
+        # app/config/services.yml
+        services:
+            # ...
+
+            _instanceof:
+                class: AppBundle\Domain\LoaderInterface
+                public: true
+                tags: ['app.domain_loader']
+
+    .. code-block:: xml
+
+        <!-- app/config/services.xml -->
+        <?xml version="1.0" encoding="UTF-8" ?>
+        <container xmlns="http://symfony.com/schema/dic/services"
+            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+            xsi:schemaLocation="http://symfony.com/schema/dic/services
+                http://symfony.com/schema/dic/services/services-1.0.xsd">
+
+            <services>
+                <!-- ... -->
+
+                <instanceof id="AppBundle\Domain\LoaderInterface" public="true">
+                    <tag name="app.domain_loader" />
+                </instanceof>
+            </services>
+        </container>
+
+    .. code-block:: php
+
+        // app/config/services.php
+        use AppBundle\Domain\LoaderInterface;
+
+        // ...
+
+        /* This method returns a child definition to define the default
+           configuration of the given class or interface */
+        $container->registerForAutoconfiguration(LoaderInterface::class)
```

Comment:
this call is **not** about the `_instanceof`. It is about registering custom logic for services using `autoconfigure`.

The new PHP DSL has a syntax to define the _instanceof conditionals. The old syntax does not, because it uses the ContainerBuilder directly and so cannot provide file-scoped features.